File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
    string txt = get_string("Text: ");

    int letters = count_letters(txt);
    int words = count_words(txt);
    int sentences = count_sentences(txt);
    float L = letters / (float)words * 100;
    float S = sentences / (float)words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    printf("%f L\n", L);
    printf("%f S\n", S);
    printf("%i letters\n", letters);
    printf("%i words\n", words + 1);
    printf("%i sentences\n", sentences);
    printf("%f index\n", index);
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char j = text[i];
        if (isalpha(j))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char j = text[i];
        if (isspace(j))
        {
            words++;
        }
    }
    return words;
}
int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char j = text[i];

        char k[] = ".?!";
        if (j == k[0] || j == k[1] || j == k[2])
        {
            sentences++;
        }
    }
    return sentences;
}

File: Week 6 Python/Problem Set 6/sentimental-readability/readability.py
# TODO
import cs50


def main():

    text = cs50.get_string('Text: ')

    # Average number of letters per 100 words in the text
    L = count_letters(text)/count_words(text)*100

    # Average number of sentences per 100 words in the text
    S = count_sentences(text)/count_words(text)*100

    # Coleman-Liau index
    index = 0.0588 * L - 0.296 * S - 15.8

    # Printing the grade while return from function
    if index < 1:
        return print("Before Grade 1")
    if index > 16:
        return print("Grade 16+")
    else:
        return print(f"Grade {round(index)}")


# Returns the number of letters in the text
def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
    return letters


# Returns the number of words in the text
def count_words(text):
    words = 0
    for i in range(len(text)):
        if text[i].isspace():
            words += 1
    return words+1


# Returns the number of sentences in the text
def count_sentences(text):
    sentences = 0
    end_chars = "!.?"
    for i in range(len(text)):
        if text[i] in end_chars:
            sentences += 1

    return sentences


main()
